sbol_utilities/component.py

```diff
@@ -36,6 +36,41 @@ def walk_tree(obj: sbol3.TopLevel):
     return {c for c in explored if isinstance(c, sbol3.Component)}
 
 
+def is_dna_part(obj: sbol3.Component) -> bool:
+    """Check if an SBOL Component is a DNA Part, i.e., having type 'SBO:DNA', and exactly 1 sequence property
+
+    :param obj: Sbol3 Document to be checked
+    :return: true if dna part
+    """
+    # must have a type of dna
+    def has_dna_type(component: sbol3.Component) -> bool:
+        return any(tyto.SBO.deoxyribonucleic_acid.is_ancestor_of(type) for type in component.types)
+
+    # there must be atleast 1 SO role, among others
+    def check_roles(component: sbol3.Component) -> bool:
+        for role in component.roles:
+            try:
+                # if tyto doesn't throw Lookup Error, role is an SO one
+                tyto.SO.get_term_by_uri(role)
+                return True
+            except LookupError:
+                pass
+        return False
+
+    # must have exactly 1 sequence property
+    def has_a_sequence_prop(component: sbol3.Component) -> bool:
+        if component.sequences:
+            return len(component.sequences) == 1
+        return False
+
+    # check all conditions if obj is an sbol3 document
+    if(isinstance(obj, sbol3.Component)):
```

Comment:
Don't need to check if it's a Component because the function already declares it is Component typed.